### DISCUSSION TITLE
Licensing (Issue moved from IEEE Gitlab CCO Issue Tracker)

### DISCUSSION THREAD

**Discussion by jimschoening1** [CommonCoreOntology/CommonCoreOntologies#409] [Announcements]
This Issue is moved her from https://opensource.ieee.org/cco/CommonCoreOntologies/-/issues/158 

Issue created 6 months ago by Alan Ruttenberg  

An issue has come up regarding IEEE mandated licenses and ontology licensing. Generally software licenses are not recommended for ontologies as ontology rights are governed solely by copyright law, whereas software licenses in addition can deal with things like patent rights.
The usual recommendation is that CC0 or CC-BY be used for ontologies.
The reason this is an issue is that CCO is destined to be used in an upcoming Foundry, and there is discussion there about what license to require of ontologies that participate. The recommendation there is CC-BY (or the CC0 public domain dedication), which is what most OBO ontologies are licensed under.
The problem has to do with derived works that derived from two sources licensed with different licenses, and the question of under what license the resulting work can be licensed. Licenses can be incompatible (as e.g. Apache and CC-BY are) and this can prevent the creation of such artifacts, despite that being a very common pattern in ontology development.
Does IEEE have a recommendation for a documentation license as separate from a software license? Such a license would be preferred for an ontology because documents are governed in the same way.
We'd really like CCO, and other IEEE ontology efforts to remain CC-BY and not land up causing downstream licensing issues that could interfere with the broadest adoption.

 =====================
Response by James Schoening
@james.schoening
· 2 months ago
Reporter
We can raise this issue to IEEE leadership, but let's explore here first, then seek full OSWG inputs and consensus.

When these standards are approved, the open-source ontology will be available for free here at https://opensource.ieee.org/cco/CommonCoreOntologies as it is now, with whatever copyright and open-source license is has. The approved standards 'document' will be available for sale as normal.

IEEE Opensource gave us BSD-3 Clause, but we can ask for another one. So let's give this a full review, seek input from the entire OSWG, and then submit our request with solid justification.

I note the CCOv1.4 Glossary at https://github.com/CommonCoreOntology/CommonCoreOntologies/blob/master/documentation/Common%20Core%20Ontology%20Glossary%201.4.xlsx has no copyright or license. Should it?

I used CCOv1.3 Glossary, plus added rows from Cyber and Person ontologies to create the Personal Data Model Glossary posted at https://github.com/standard-personal-data-model/PDM-Glossary (for the mydata.org community). Each cell has a link to the source ontology, but do I also need to cite copyright and license? If so, it is different for each row. If needed I could add a new column for copyright and license so each row could cite this.

Response by: 
Alan Ruttenberg
@alanruttenberg
· 1 month ago
Author
Developer
#1, OK

#2, the ontologies should also be available at their current locations ("don't change IRIs") e.g. http://www.ontologyrepository.com/CommonCoreOntologies/Mid/MergedAllCoreOntology We will arrange that with the PURLs.

#3 OK

#4 Yes. CCO uses an ontology annotation property. It's not really consistent across versions. In an earlier version they used two properties in CCO: "content license" which was set to https://creativecommons.org/licenses/by/3.0/ and "code license" which was set to https://opensource.org/licenses/BSD-3-Clause

In 1.4 they use dc:license with the value https://opensource.org/licenses/BSD-3-Clause Consistent with my recommendation would be to use dc:license with value https://creativecommons.org/licenses/by/4.0/

To add an ontology annotation in Protege use the "Active Ontology" tab. There's a "+" sign beside the label "Annotations".

#5 The URL you give for that is to a github repository. Good practice is to have a file called LICENSE with the license, or to use Github's method for attaching a license to a repository

For the Excel file, the content is from other sources so you have to follow the licensing requirements for those sources.

For instance BSD says "Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer"...

CC-BY 4 says: Attribution — You must give appropriate credit , provide a link to the license, and indicate if changes were made . You may do so in any reasonable manner, but not in any way that suggests the licensor endorses you or your use.

**Comment by johnbeve** [comment]
Moved from previous thread:  

From [BrendaBraitling](https://github.com/BrendaBraitling): "What is developed - other than patent/IP materials - is IEEE's property or it holds a favorable license to use.. And for indemnification purposes should be under the control of an officer/IEEE member of the WG... perhaps the OS Project Leader, if it is being managed as an OS project.

If the WG is generating expenses it should have a Treasurer and a method of handling those expenses - even if it is via donation. Any use of meeting or communication technologies, hosting, etc. should be transparent and auditable.
IEEE has AudCom and ethics specialists to help resolve these sorts of things.

Domain names point to things - so it may be possible to have a primary with a failover or mirror backend configuration. It decouples the point of reference from the physical server system. It frees the WG to select any backend and does not disrupt coding that is already in use.

RE: PURL at IEEE, My understanding is that it is newly in production jut out of testing and piloting... SO my concern would be reliability. Ontologies are important to AI and may become quite heavily accessed."

**Comment by alanruttenberg** [comment]
From where does the conclusion that what is developed is IEEE's property?

Copyright is not automatically assigned. If IEEE wanted to "own" contributions to CCO it would need some means of gaining consent of the authors to assign their copyright to them. Sometimes this is done with a contributor agreement. However, I do not recall being asked for such consent. Moreover CUBRC owns the copyright to thus-far developed CCO and would explicitly have to assign copyright for IEEE to have it.

What the comment does point out is that we need to have an explicit policy on how copyright is to work, and on getting explicit consent for authors of e.g. contributed terms, to assign copyright, if that's what we want.

Not sure where the @BrendaBraitling comment originates from, but I may not yet have happened upon it in my email stream. Will comment directly when I find out.

**Comment by johnbeve** [comment]
@alanruttenberg Brenda had commented on @jimschoening1's posts from https://github.com/CommonCoreOntology/CommonCoreOntologies/issues/105#issuecomment-2266747290. I moved her comment here since the previous issue was closed and her comment was picking up on the discussion here.
